src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java

```diff
@@ -123,9 +124,11 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
 
         final String nodeName = buildData.getNodeName("").isEmpty() ? pipelineData.getNodeName("") : buildData.getNodeName("");
         buildSpan.setTag(CITags.NODE_NAME, nodeName);
-
-        final String nodeHostname = buildData.getHostname("").isEmpty() ? pipelineData.getHostname("") : buildData.getHostname("");
-        buildSpan.setTag(CITags._DD_HOSTNAME, nodeHostname);
+        // If the NodeName == master, we don't set _dd.hostname. It will be overridden by the Datadog Agent. (Traces are only available using Datadog Agent)
+        // If the NodeName != master, we set _dd.hostname to 'none' explicitly, cause we cannot calculate the worker hostname.
+        if(!"master".equalsIgnoreCase(nodeName)) {
+            buildSpan.setTag(CITags._DD_HOSTNAME, HOSTNAME_NONE);
```

Comment:
What's the impact of setting `none` here? This has a pretty big impact for metrics for instance (as it would create a new host with the "none" name), but this should be for traces only in that situation. Are we ok with setting a dummy value?

Reply:
As far as we know, APM doesn't create hosts based on the `_dd.hostname` tag. In this case, the value `none` is a special value for APM that is excluded from per-host billing. Additionally, if we don't set it and we let the agent put the master hostname, it'd be worse from a user perspective because we will be showing wrong information.

Reply:
Perfect, I didn't know about billing implications